packages/vm/src/cache.rs

```diff
@@ -53,6 +53,14 @@ pub struct Metrics {
     pub size_memory_cache: usize,
 }
 
+#[derive(Debug, Clone)]
+pub struct HeavyMetrics {
+    // It is *intentional* that this is only a vector
+    // We don't need a potentially expensive hashing algorithm here
+    // The checksums are sourced from a hashmap already, ensuring uniqueness of the checksums
+    pub hits_per_pinned_contract: Vec<(Checksum, u32)>,
```

Comment:
On the `wasmvm` side, we could serialize this with a format like MessagePack, so we don't have to juggle this stuff around the FFI boundary.

Reply:
Yeah, we'll look into that later on. libwasmvm (the Rust project in wasmvm) can do any transformation we need.